### DISCUSSION TITLE
Getting Form annotation values

### DISCUSSION THREAD

**Discussion by chefinDan** [wojtekmaj/react-pdf#1469] [Q&A] [answered]
In Mozilla's documentation they mention that form inputs are stored in a sandbox built using Quickjs. This seems to be the case for the generic pdf_viewer build at least.
After looking through your source it looks like you are not using annotation storage for form values. What then would be the recommended method for handling form values? 
I've had some success just querying all the input elements, but this seems less than ideal. 
Have any thoughts?

**Comment by wojtekmaj** [comment, marked as answer]
There's no implementation of annotation storage of any kind, so what you're doing at the moment is the best thing you can do.

But if you think there's a way to make AnnotationLayer more... Reactive, I'm open to suggestions!

**Reply by chefinDan** [reply]
Wow thank you for the prompt response!
I have also had some luck embedding Mozilla's generic pdf_viewer as an iframe within my react app. This modifies the window object and I generally try to avoid that, but if Mozilla does it then it must be ok right?
I'll update here if/when I find an alternative.

**Reply by wojtekmaj** [reply]
My younger self would hate that, now my stance on that is: whatever makes your app work ans makes you happy. 😅
